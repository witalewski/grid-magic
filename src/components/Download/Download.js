import React, { Component } from 'react';
import { object, number } from 'prop-types';
import { observer, inject } from 'mobx-react';

class Download extends Component {
  static propTypes = {
    imageProcessor: object.isRequired,
    previewCanvas: object.isRequired,
    tileSize: number.isRequired,
  };

  render() {
    return (
        <button onClick={() => this.props.imageProcessor.downloadImages(this.props.previewCanvas,this.props.tileSize)}>Download images</button>
      );
  }
}

export { Download };
export default inject(({ appState }) => ({
  imageProcessor: appState.imageProcessor,
  previewCanvas: appState.previewCanvas,
  tileSize: appState.tileSize,
}))(observer(Download));
