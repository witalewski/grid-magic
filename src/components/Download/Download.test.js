import React from 'react';
import { Download } from './Download';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Download', () => {
  it('displays download', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Download downloadImages={jest.fn()} />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
