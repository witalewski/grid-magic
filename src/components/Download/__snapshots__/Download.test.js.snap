// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download displays download 1`] = `
<button
  className="download-button"
  onClick={[MockFunction]}
>
  Download images
</button>
`;
